thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1741,12 +1770,16 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
         Owned<IEngineRowAllocator> fetchDiskAllocator;
         CThorContiguousRowBuffer prefetchSource;
         std::vector<PartIO> partIOs;
+        std::unique_ptr<std::vector<std::atomic_bool>> partIOCreated;
 
         inline const PartIO &queryFetchPartIO(unsigned selected, unsigned partNo, unsigned copy, bool compressed, bool encrypted)
         {
             PartIO &partIO = partIOs[selected];
-            if (!partIO.stream)
+            if (!(*partIOCreated)[selected])
```

Comment:
this change isn't needed. The process() thread is the only thread that updates partIOs